### PR TITLE
Users: Default to en_US locale, UTC time zone

### LINKED ISSUE #34: Default locale/timezone are vague
If a user doesn't have a locale or timezone setup, the dropdown boxes default to `af-ZA` and `Africa/Abidjan` respectively.
Maybe it's a good idea to put an empty value as option for these fields.

### PR DESCRIPTION
**Pull request by nphilipp**
If the user is added, these fields will be set to the defaults if unset. If an existing user is modified and one of these fields is unset, pull the existing IPA user and use its information for unset fields.

Fixes: #34

### CODE REVIEW

**Comment by nphilipp** [comment]
The latest push attempts to do the right thing for existing IPA users with unset locale or time zone.

File: fas2ipa/users.py [hunk 1]
```diff
@@ -354,7 +356,14 @@ def migrate_user(self, person):
...
                         del user_args["full_name"]
 
-                    # Avoid overwriting already set fields
+                    # Avoid resetting already set fields
+                    if user_args["faslocale"] is None or user_args["fastimezone"] is None:
+                        ipa_user = self.user_show(username)
+                        if not user_args["faslocale"]:
+                            user_args["faslocale"] = ipa_user.get("faslocale") or "en_US"
+                        if not user_args["fastimezone"]:
+                            user_args["fastimezone"] = ipa_user.get("fastimezone") or "UTC"
```
Comment (abompard) [inline fas2ipa/users.py:365]: I don't understand the reason for this block: if the user has already been imported (say, in the Fedora run) and we get a user with the same username in a second import (say, in CentOS) that did not set their locales & timezones, then these values will be `None` and they will be removed by the dict loop right below, no?

Comment (nphilipp) [reply fas2ipa/users.py:365]: This reflects that users could have been created in IPA with an unset locale, time zone in previous runs (e.g. staging): When modifying an existing IPA user, only apply the defaults to these fields if they're unset in the IPA user, too, not just the processed FAS user.

Comment (abompard) [reply fas2ipa/users.py:365]: Ah, I see. Indeed in production all users will already exist. I'm a bit afraid that adding the `user_show` call for all existing user (thus all users) will make the import much slower.

Maybe we can run a post-import script to get all users with unset locale & timezone and set the defaults?

Comment (nphilipp) [reply fas2ipa/users.py:365]: The `user_show()` call will only be run for the subset of (FAS) users that have an unset locale or time zone (line 360), this affects ~8% of users.

Comment (abompard) [reply fas2ipa/users.py:365]: Right, it should be fine :-)

**Review by abompard — APPROVED** [review]

**Pull request merged**